### PR TITLE
[SPARK-38740][SQL][TESTS] Test the error class: INVALID_JSON_SCHEMA_MAPTYPE

### PR DESCRIPTION
**Pull request by panbingkun**
## What changes were proposed in this pull request?
This PR aims to add a test for the error class INVALID_JSON_SCHEMA_MAPTYPE to `QueryCompilationErrorsSuite`.

### Why are the changes needed?
The changes improve test coverage, and document expected error messages in tests.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By running new test:
```
$ build/sbt "sql/testOnly *QueryCompilationErrorsSuite*"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 1]
```diff
@@ -364,6 +364,21 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
...
+    withTempPath { path =>
+      val schema = StructType(
+        StructField("map", MapType(IntegerType, IntegerType, true), false) :: Nil)
+      val e = intercept[AnalysisException] {
+        spark.read.schema(schema).json(path.getCanonicalPath)
+      }
+      assert(e.getErrorClass === "INVALID_JSON_SCHEMA_MAPTYPE")
+      assert(e.getMessage ===
+        "Input schema StructType(StructField(map,MapType(IntegerType,IntegerType,true),false)) " +
+          "can only contain StringType as a key type for a MapType.")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:379]: Could you use the `checkErrorClass` method added by https://github.com/apache/spark/pull/36287.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 2]
```diff
@@ -380,6 +380,24 @@ class QueryCompilationErrorsSuite
...
+
+  test("INVALID_JSON_SCHEMA_MAPTYPE: " +
+    "Parse JSON rows can only contain StringType as a key type for a MapType.") {
+    withTempPath { path =>
+      val schema = StructType(
+        StructField("map", MapType(IntegerType, IntegerType, true), false) :: Nil)
+
+      checkErrorClass(
+        exception = intercept[AnalysisException] {
+          spark.read.schema(schema).json(path.getCanonicalPath)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:392]: I think no need to create a temporary dir. AFAK, `json()` accepts a `Dataset`. Could you pass an empty dataset to `json()`, please.

**Comment by MaxGekk** [comment]
@panbingkun Could you resolve conflicts, please.

**Comment by panbingkun** [comment]
> @panbingkun Could you resolve conflicts, please.

Done

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**